tokio/src/io/read_buf.rs

```diff
@@ -55,6 +59,47 @@ impl<'a> ReadBuf<'a> {
         }
     }
 
+    cfg_io_util! {
+         fn from_buf(buf: &'a mut impl bytes::BufMut) -> ReadBuf<'a> {
+            let buf = buf.chunk_mut();
+            ReadBuf {
+                // SAFETY: ReadBuf guarantees that no uninitialized bytes are written to `buf`
+                // (same guarantee as `UninitSlice`)
+                buf: unsafe {
+                    std::slice::from_raw_parts_mut(buf.as_mut_ptr() as *mut MaybeUninit<u8>, buf.len())
+                },
+                filled: 0,
+                initialized: 0,
+            }
+        }
+
+        #[doc(hidden)]
+        pub fn with_buf<R>(
+            buf: &'a mut impl bytes::BufMut,
+            f: impl FnOnce(&mut ReadBuf<'_>) -> Poll<io::Result<R>>,
+        ) -> Poll<io::Result<R>> {
```

Comment:
We don't do this. It should either be public and non-hidden, or it should be defined in `tokio-util`.

Reply:
I can do either, but it is used in both `tokio` and `tokio-util` so it would have to be duplicated or made public.

Reply:
I would be ok with making both of these methods public. They would need to be documented.